pandas/tests/indexing/test_indexing.py

```diff
@@ -529,13 +529,19 @@ def test_astype_assignment(self):
         expected = DataFrame(
             [[1, 2, "3", ".4", 5, 6.0, "foo"]], columns=list("ABCDEFG")
         )
+        # original (object) array can hold new values, so setting is inplace
+        expected["A"] = expected["A"].astype(object)
+        expected["B"] = expected["B"].astype(object)
```

Comment:
I think this is an undesired change in behaviour .. (although consistent with the rules). It's rather clear that when doing `df.iloc[:, 0:2] = df.iloc[:, 0:2].astype(np.int64)` you want to change it to int64 ..